counterparty-rs/src/b58.rs

```diff
@@ -1,45 +1,42 @@
-use pyo3::prelude::*;
-use pyo3::types::PyBytes;
 use pyo3::exceptions::PyValueError;
+use pyo3::prelude::*;
 
 #[pyfunction]
 fn b58_encode(decoded: &[u8]) -> String {
     bs58::encode(decoded).with_check().into_string()
 }
 
 #[pyfunction]
-fn b58_encode_list(decoded_list: Vec<&[u8]>) -> Vec<String> {
+fn b58_encode_list(decoded_list: Vec<Vec<u8>>) -> Vec<String> {
     decoded_list.iter().map(|x| b58_encode(x)).collect()
 }
 
 #[pyfunction]
-fn b58_decode<'p>(py: Python<'p>, encoded: &str) -> PyResult<&'p PyBytes> {
-    let decoded = bs58::decode(encoded)
-        .with_check(None)
-        .into_vec();
+fn b58_decode(encoded: &str) -> PyResult<Vec<u8>> {
+    let decoded = bs58::decode(encoded).with_check(None).into_vec();
 
     match decoded {
-        Ok(s) => Ok(PyBytes::new(py, &s)),
+        Ok(s) => Ok(s),
         Err(_) => Err(PyValueError::new_err("Bad input")),
     }
 }
 
 #[pyfunction]
-fn b58_decode_list<'p>(py: Python<'p>, encoded_list: Vec<&str>) -> PyResult<Vec<&'p PyBytes>> {
+fn b58_decode_list(encoded_list: Vec<String>) -> PyResult<Vec<Vec<u8>>> {
```

Comment:
Using rust types here's significantly simplifies the code as we don't have to do any casting or introduce lifetime parameters.